pkg/condition/condition.go

```diff
@@ -0,0 +1,267 @@
+package condition
+
+// adapted from rancher/wrangler
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Cond string
+
+func (c Cond) GetStatus(obj interface{}) string {
+	return getStatus(obj, string(c))
+}
+
+func (c Cond) SetError(obj interface{}, reason string, err error) {
+	if err == nil {
+		c.True(obj)
+		c.Message(obj, "")
+		c.Reason(obj, reason)
+		return
+	}
+	if reason == "" {
+		reason = "Error"
+	}
+	c.False(obj)
+	c.Message(obj, err.Error())
+	c.Reason(obj, reason)
+}
+
+func (c Cond) MatchesError(obj interface{}, reason string, err error) bool {
+	if err == nil {
+		return c.IsTrue(obj) &&
+			c.GetMessage(obj) == "" &&
+			c.GetReason(obj) == reason
+	}
+	if reason == "" {
+		reason = "Error"
+	}
+	return c.IsFalse(obj) &&
+		c.GetMessage(obj) == err.Error() &&
+		c.GetReason(obj) == reason
+}
+
+func (c Cond) SetStatus(obj interface{}, status string) {
+	setStatus(obj, string(c), status)
+}
+
+func (c Cond) SetStatusBool(obj interface{}, val bool) {
+	if val {
+		setStatus(obj, string(c), "True")
+	} else {
+		setStatus(obj, string(c), "False")
+	}
+}
+
+func (c Cond) True(obj interface{}) {
+	setStatus(obj, string(c), "True")
+}
+
+func (c Cond) IsTrue(obj interface{}) bool {
+	return getStatus(obj, string(c)) == "True"
+}
+
+func (c Cond) False(obj interface{}) {
+	setStatus(obj, string(c), "False")
+}
+
+func (c Cond) IsFalse(obj interface{}) bool {
+	return getStatus(obj, string(c)) == "False"
+}
+
+func (c Cond) Unknown(obj interface{}) {
+	setStatus(obj, string(c), "Unknown")
+}
+
+func (c Cond) IsUnknown(obj interface{}) bool {
+	return getStatus(obj, string(c)) == "Unknown"
+}
+
+func (c Cond) LastUpdated(obj interface{}, ts string) {
+	setTS(obj, string(c), ts)
+}
+
+func (c Cond) GetLastUpdated(obj interface{}) string {
+	return getTS(obj, string(c))
+}
+
+func (c Cond) GetLastTransitionTime(obj interface{}) time.Time {
+	return getLastTransitionTime(obj, string(c))
+}
```

Comment:
This entire file is a copy from rancher/wrangler with the addition of `GetLastTransitionTime` because there seems to be no lib out there for interacting with conditions (other than norman or wrangler).